File: 2-PERIODO/AEDS-II/trabalhos/tt02/cometa.c
#include <stdio.h>
#include <string.h>

int isFim(char *s) {
  int length = strlen(s);
  return (length == 1 && s[0] == '0');
}

int cometa(char *s) {
  int i, anoAtual, proximoAno = 1986;
  i = anoAtual = 0;
  while (i < strlen(s)) {
    anoAtual = (anoAtual * 10) + (s[i] - '0');
    i++;
  }
  if ((anoAtual - 1986) % 76 == 0) {
    proximoAno = anoAtual + 76;
  } else {
    while (proximoAno < anoAtual) {
      proximoAno += 76;
    }
  }
  return proximoAno;
}

int main() {
  int numEntrada = 0;
  char arrayStrings[2000][1000];

  do {
    fgets(arrayStrings[numEntrada], 1000, stdin); // Leitura da linha
    setbuf(stdin, NULL);
    arrayStrings[numEntrada][strcspn(arrayStrings[numEntrada], "\n")] =
        0; // Retira o caractere de '/n' do fim da string
  } while (isFim(arrayStrings[numEntrada++]) ==
           0); // Checa se a linha é igual a "FIM"
  numEntrada--;

  for (int i = 0; i < numEntrada; i++) {
    printf("%d\n", cometa(arrayStrings[i]));
  }
}

File: 2-PERIODO/AEDS-II/trabalhos/tt02/cometa.java
class cometa {
  public static boolean isFim(String s){
    return (s.length() == 1 && s.charAt(0) == '0');
  }

  public static int proxAno(String s){
    int i, anoAtual, proximoAno = 1986;
    i = anoAtual = 0;
    while (i < s.length()) {
      anoAtual = (anoAtual * 10) + (s.charAt(i) - '0');
      i++;
    }
    if ((anoAtual - 1986)%76 == 0) {
      proximoAno = anoAtual + 76;
    } else {
      while (proximoAno < anoAtual) {
        proximoAno += 76;
      }
    }
    return proximoAno;
  }


  public static void main (String[] args){
    String[] entrada = new String[1000];
    int numEntrada = 0;

    //Leitura da entrada padrao
    do {
      entrada[numEntrada] = MyIO.readLine();
    } while (isFim(entrada[numEntrada++]) == false);
    numEntrada--;   //Desconsiderar ultima linha contendo a palavra FIM

    //Para cada linha de entrada, gerando uma de saida contendo o numero de letras maiusculas da entrada
    for(int i = 0; i < numEntrada; i++){
      MyIO.println(proxAno(entrada[i]));
    }
  }
}
